torch/csrc/distributed/rpc/PythonRpcHandler.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <torch/csrc/distributed/rpc/Message.h>
+#include <torch/csrc/utils/pybind.h>
+#include <Python.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+class PythonRpcHandler {
+public:
+  // initialize python interpreter if not yet; initialize python module object
+  // and function objects
+  static void init();
+  // clean up resources if py_initialize() was called; free python objects
+  static void cleanUp();
+
+  // execute user defined python function, result is pickled to binary string
+  static std::vector<char> generatePythonUDFResult(const Message& request);
+  // returned python UDF result is pickled binary string, so run python
+  // function to unpickle the python UDF result and return pyObject to user
+  static py::object loadPythonUDFResult(const Message& message);
+
+private:
+  static PyObject* module_;
+  static PyObject* runUDFFunction_;
+  static PyObject* loadResultFunction_;
```

Comment:
For new code we are writing, there's NO reason to use raw `PyObject*` pointers. The manual refcounting is an endless bugfarm. Please use `THPObjectPtr`, or you can also use the pybind11 equivalent if you like.